This patch provides a workaround to one long-standing problem with building
extension modules on Cygwin; see https://bugs.python.org/issue2445 for more
background.
diff --git a/Lib/distutils/unixccompiler.py.orig b/Lib/distutils/unixccompiler.py
index 3f321c2..fb5a8fc 100644
--- a/Lib/distutils/unixccompiler.py
+++ b/Lib/distutils/unixccompiler.py
@@ -81,6 +81,7 @@ class UnixCCompiler(CCompiler):
     xcode_stub_lib_format = dylib_lib_format
     if sys.platform == "cygwin":
         exe_extension = ".exe"
+        dylib_lib_extension = ".dll.a"
 
     def preprocess(self, source, output_file=None, macros=None,
                    include_dirs=None, extra_preargs=None, extra_postargs=None):
